project/Dependencies.scala

```diff
@@ -27,7 +27,7 @@ object Dependencies {
 
   val scala212Version = "2.12.15"
   val scala213Version = "2.13.8"
-  val scala3Version = "3.1.1"
+  val scala3Version = "3.1.2"
```

Comment:
I see some improvements in compile times with this, although not nearly enough to offset the need for the `tailSwitchN` optimisatins